riocli/deployment/update.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2023 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from concurrent.futures import ThreadPoolExecutor
+
+import click
+from click_help_colors import HelpColorsCommand
+
+from rapyuta_io import Client, DeploymentPhaseConstants
+
+from riocli.config import new_client
+from riocli.constants import Symbols, Colors
+from riocli.utils import tabulate_data
+from riocli.utils.spinner import with_spinner
+
+@click.command(
+    'update',
+    cls=HelpColorsCommand,
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.option('--force', '-f', '--silent', is_flag=True, default=False,
+              help='Skip confirmation')
+@click.option('--update-all', '-a', is_flag=True, default=False,
+              help='Updates all deployments')
+@click.argument('deployment-name-or-regex', type=str, default="")
+@click.argument('workers', type=int, default=6)
+@with_spinner(text="Updating deployment...")
+def update_deployment(force: bool, deployment_name_or_regex: str, update_all: bool, workers: int, spinner=None) -> None:
+    """
+    Delete the deployment from the Platform
+    """
+    client = new_client()
+    if not(deployment_name_or_regex or update_all):
+        spinner.text = "Nothing to update"
+        spinner.green.ok(Symbols.SUCCESS)
+        return
+
+    try:
+        deployments = fetch_deployments(client, deployment_name_or_regex, update_all)
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to update deployment(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+    if not deployments:
+        spinner.text = "Nothing to update"
+        spinner.green.ok(Symbols.SUCCESS)
+        return
+
+    headers = ["name", "guid", "phase"]
+    data = [[deployment.name, deployment.deploymentId, deployment.phase] for deployment in deployments]
+
+    spinner.write("")
+    tabulate_data(data, headers)
+
+    if not force:
+        with spinner.hidden():
+            click.confirm("Do you want to update above deployment(s)?", default=True, abort=True)
+    try:
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            executor.map(_apply_update, [client] * len(deployments), deployments)
```

Comment:
Instead of doing the `len()` stuff, you can use `partial` here
```suggestion
        func = functools.partial(_apply_update, client)
        with ThreadPoolExecutor(max_workers=workers) as executor:
            executor.map(func, deployments)
```